datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -223,6 +226,15 @@ public class Coordinator implements ZkAdapter.ZkAdapterListener, MetricsAware {
   private Future<?> _leaderDoAssignmentScheduledFuture = null;
   private volatile boolean _zkSessionExpired = false;
 
+  // Cache all the throughput violating topics per datastream
```

Comment:
Can you please confirm that no 2 streams can have the same name, even across different connectors? If that's not the case might be worth storing Connector + Stream Name tuple as the key here.

Reply:
I think we can safely assume that datastream names are unique across a cluster. I am using a map with the key as the datastream name here since we'll be storing this information within ZK similarly under the /DMS/ path with datastream as the key.